src/crud_ops.rs

```diff
@@ -143,6 +225,51 @@ impl Client {
         counter!(metrics::total_delete_ops).increment(1);
         with_retries!(self, self.delete_impl(path).await)
     }
+
+    async fn bulk_delete_impl(&self, paths: &Vec<Path>) -> crate::Result<Vec<(Path, crate::Error)>> {
+        let stream = stream::iter(paths.iter().map(|path| Ok(path.clone()))).boxed();
+        let indexed_results = self.store.delete_stream(stream)
+            .enumerate()
+            .filter_map(|(index, result)| async move {
+                Some((index, result))
+            }).collect::<Vec<(usize, Result<Path, object_store::Error>)>>().await;
+        // We need to count the number of callbacks called to determine if we need to raise an error
+        // if some paths were not processed at all.
+        let callbacks_called = indexed_results.len();
+        let bulk_failed_entries = indexed_results.into_iter().filter_map(|(index, result)| {
```

Comment:
The `enumerate` should be here:
```suggestion
        let bulk_failed_entries = results
            .into_iter()
            .enumerate()
            .filter_map(|(index, result)| {
```

Reply:
Also, I don't think the name `bulk_failed_entries` applies anymore (as that moved to set_payload). Maybe just `failures` is clear enough.